packages/metro-config/src/index.js

```diff
@@ -27,28 +29,61 @@ function defaultWatchFolders(projectRoot) {
   // the same monorepo as the target package.
   const thisPackage = path.dirname(findPackage(projectRoot) || "");
 
+  /** @type {RepoPackageInfo} */
+  const info = { externalPackages: [], internalPackages: [] };
   try {
     const root = getWorkspaceRoot(thisPackage);
     if (!root) {
-      return [];
+      return info;
     }
 
     const packages = getAllPackageJsonFiles(thisPackage);
     if (!Array.isArray(packages) || packages.length === 0) {
-      return [];
+      return info;
     }
 
     // In a monorepo, in particular when using Yarn workspaces, packages are
-    // symlinked in the root `node_modules` folder so it needs to be watched.
-    return [
-      path.join(root, "node_modules"),
-      ...packages.map((pkg) => path.dirname(pkg)),
-    ];
+    // symlinked in the root `node_modules` folder so it needs to be included.
+    info.externalPackages.push(path.join(root, "node_modules"));
+
+    info.internalPackages = packages.map((pkg) => path.dirname(pkg));
+
+    return info;
   } catch (_) {
-    return [];
+    return info;
   }
 }
 
+/**
+ * A minimum list of folders that should be watched by Metro.
+ * @param {string | undefined} projectRoot
+ * @returns {string[]}
+ */
+function defaultWatchFolders(projectRoot) {
+  const info = getRepoPackageInfo(projectRoot);
+  return [...info.externalPackages, ...info.internalPackages];
+}
+
+/**
+ * A minimum list of in-repo folders that should be watched by Metro.
+ * @param {string | undefined} projectRoot
+ * @returns {string[]}
+ */
+function defaultWatchFoldersInRepo(projectRoot) {
+  const info = getRepoPackageInfo(projectRoot);
+  return info.internalPackages;
+}
+
+/**
+ * A minimum list of package root folders.
+ * @param {string | undefined} projectRoot
+ * @returns {string[]}
+ */
+function defaultRoots(projectRoot) {
+  const info = getRepoPackageInfo(projectRoot);
+  return [...info.externalPackages, ...info.internalPackages];
```

Comment:
The roots should already contain the packages:
```suggestion
  const info = getRepoPackageInfo(projectRoot);
  return info.externalPackages;
```

Reply:
They don't. `externalPackages` is stuff that's not in-repo, `internalPackages` is only in-repo stuff.